src/pcm.c

```diff
@@ -1531,20 +1546,25 @@ static int pcm_mmap_transfer(struct pcm *pcm, void *buffer, unsigned int frames)
     }
 
     while (frames) {
+	pcm_sync_ptr(pcm, 0);
         avail = pcm_mmap_avail(pcm);
 
-        if (!avail) {
+        if (avail < pcm->config.avail_min) {
+            int time = -1;
             if (pcm->flags & PCM_NONBLOCK) {
                 errno = EAGAIN;
                 break;
             }
+	    if (pcm->flags & PCM_NOIRQ)
```

Comment:
Do we still need this `if`?

Reply:
According to the history https://github.com/tinyalsa/tinyalsa/commit/73b9c679a656c7b0f5e265dae5a76664c7d03031
It's to fix the wait timeout issue for NOIRQ case, so here is an if.

And the time calculation is also based on noirq_frames_per_msec. so we need this if.
https://github.com/tinyalsa/tinyalsa/pull/259/files#diff-3f4d00bbfd80c7c121a953205cb538e67d28f3a9a1c017e4b6e5fd66ee038afeR1559